generation/localizedstringkit/detection.py

```diff
@@ -148,21 +149,35 @@ def _detect_strings(
                         bundle="LocalizedStringKit.bundle",
                     )
                 )
-            elif match[-1] == bundle_pattern:
-                # Localized call with custom bundle
-                results.append(
-                    LocalizedString(
-                        key=None,
-                        value=match[0],
-                        language="en",
-                        table="LocalizedStringKit",
-                        comment=match[1],
-                        key_extension=None,
-                        bundle=match[2],
+            elif length == 3:
+                if match[-1] == bundle_pattern:
+                    # Localized call with custom bundle
+                    results.append(
+                        LocalizedString(
+                            key=None,
+                            value=match[0],
+                            language="en",
+                            table="LocalizedStringKit",
+                            comment=match[1],
+                            key_extension=None,
+                            bundle=match[2],
+                        )
+                    )
+                else:
+                    # Localized call with key extension
+                    results.append(
+                        LocalizedString(
+                            key=None,
+                            value=match[0],
+                            language="en",
+                            table="LocalizedStringKit",
+                            comment=match[1],
+                            key_extension=match[2],
+                            bundle="LocalizedStringKit.bundle",
+                        )
                     )
-                )
             else:
```

Comment:
I would validate `match` length before enumerating the values for safety sake - in a follow up